File: lab2/zadanie3/zadanie3.py
def wypisz(dzielniki, n):
	elementy = []
	print(f"{n}=",end="")
	for d, potega in dzielniki.items():
		if potega:
			if potega > 1:
				elementy.append(f"{d}^{potega}")
			else:
				elementy.append(f"{d}")
	print("*".join(elementy))


def rozklad(n, dzielniki):
	d = 3
	dd = 9
	potega = 0

	while n%2 == 0:
		potega += 1
		n = n // 2
	dzielniki[2] = potega
	
	while n >= dd:
		potega = 0
		while n%d == 0:
			n = n // d
			potega += 1
		dzielniki[d] = potega

		dd = dd + 4*d +4
		d = d + 2
	if n > 1:
		dzielniki[n] = 1

if __name__=="__main__":
	n = int(input("Podaj liczbę: "))
	dzielniki = {}
	rozklad(n, dzielniki)
	wypisz(dzielniki, n)

File: lab2/zadanie3/zadanie3.c
#include <stdio.h>
#include <stdlib.h>

int n;

void wypisz(int n, int tab[]){
	int pierwszy = 1;

	printf("%d=", n);
	for (int i=0; i < n; i ++){
		if (tab[i]){
			if (!pierwszy) {
				printf("*");
			}
			else {
				pierwszy = 0;
			}

			if (tab[i] > 1){
				printf("%d^%d", (i+2), tab[i]);
			}
			else {
				printf("%d", (i+2));
			}
		}
	}
}

void rozklad(int n, int tab[]){
	int d = 3;
	int dd = 9;

	while (n%2 == 0){
		tab[0] ++;
		n = n / 2;
	}

	while (n >= dd){
		while (n%d == 0){
			n = n / d;
			tab[d-2] ++;
		}

		dd = dd + 4*d + 4;
		d = d + 2;
	}
	if (n > 1){
			tab[n-2] = 1;
	}
	
}

int main(){
	int *tab;
	printf("Podaj liczbę: ");
	scanf("%d", &n);
	tab = malloc(sizeof(int)*(n-1));
	for (int i=0; i < n; i++){
		tab[i]=0;
	}
	rozklad(n, tab);
	wypisz(n, tab);

}
